src/hardware/disk_noise.h

```diff
@@ -0,0 +1,75 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2025-2025  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_DISK_NOISE_H
+#define DOSBOX_DISK_NOISE_H
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "mixer.h"
+
+class DiskNoiseDevice {
+public:
+	DiskNoiseDevice(const bool enable_disk_noise,
+	                const std::string& spin_up_sample_path,
+	                const std::string& spin_sample_path,
+	                const std::vector<std::string>& seek_sample_paths,
+	                const float& spin_volume, const float& seek_volume,
+	                bool loop_spin_sample);
+
+	void ActivateSpin();
+	void PlaySeek();
+	void Shutdown();
+
+private:
+	const unsigned int SampleRate = 22050;
+	bool enable_disk_noise_       = false;
+
+	float spin_volume_ = 0.0;
+	float seek_volume_ = 0.0;
+	std::vector<float> spin_up_sample_;
+	std::vector<float> spin_sample_;
+	std::vector<float> current_seek_sample_;
+	std::vector<std::vector<float>> seek_samples_;
+	std::vector<int> seek_sample_weights_;
+
+	bool loop_spin_sample_ = false;
+	size_t spin_pos_    = 0;
+	size_t spin_up_pos_ = 0;
+	size_t seek_pos_    = 0;
```

Comment:
Suggest to use nested structs instead (e.g. one for spin, another for seek, etc.)

There are lots of examples for this style we prefer in `mixer.h`:

https://github.com/dosbox-staging/dosbox-staging/blob/94ff0ca45ea88b61eeada4944c953da390599ff1/include/mixer.h#L415-L461